homeassistant/components/hyundai_kia_connect/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for Hyundai / Kia Connect integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from hyundai_kia_connect_api import Token, get_implementation_by_region_brand
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_REGION,
+    CONF_TOKEN,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import BRANDS, CONF_BRAND, DEFAULT_PIN, DOMAIN, REGIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_PIN): str,
+        vol.Required(CONF_REGION): vol.In(REGIONS),
+        vol.Required(CONF_BRAND): vol.In(BRANDS),
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, user_input: dict[str, Any]) -> Token:
+    """Validate the user input allows us to connect."""
+
+    username = user_input[CONF_USERNAME]
+    password = user_input[CONF_PASSWORD]
+    region = user_input[CONF_REGION]
+    pin = user_input.get(CONF_PIN, DEFAULT_PIN)
+    brand = user_input[CONF_BRAND]
+
+    api = get_implementation_by_region_brand(region, brand, username, password, pin)
+    token: Token = await hass.async_add_executor_job(api.login)
+
+    if token is None:
+        raise InvalidAuth
+
+    return token
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hyundai / Kia Connect Bluelink."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        await self.async_set_unique_id(DOMAIN)
```

Comment:
Why limit it by domain?

Reply:
as a starting point, I prefer to set one instance of integration and add multi car support as later on. but it is definitely on roadmap

Reply:
It is more difficult to change that later, as you would have to migrate all existing entities. I would not recommend on doing this, unless absolutely not possible otherwise (which, is not the case as far as I can see)

Reply:
will address in future commit, thanks for callout

Reply:
What is blocking it to address it in this PR? I don't see the limitation currently, what causes this limitation to be needed?

Reply:
I mean, under this PR, in future commit

Reply:
should be fixed now